### PR TITLE
Allow to overwrite share target mountpoints

### PR DESCRIPTION
**Pull request by nickvergessen**
Required for https://github.com/nextcloud/spreed/pull/2876

### CODE REVIEW

**Comment by nickvergessen** [comment]
Would be nice to get this in, so we can finish the Talk side of this feature.

**Review by ChristophWurst — APPROVED** [review]
Not the most appropriate use of events, but fine by me

**Review by juliusknorr — APPROVED** [review]

File: lib/public/Share/Events/VerifyMountPointEvent.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	/** @var View */
+	private $view;
+	/** @var string */
+	private $parent;
+
+	/**
+	 * @since 19.0.0
+	 */
+	public function __construct(IShare $share,
+								View $view,
```
Comment (rullzer) [inline lib/public/Share/Events/VerifyMountPointEvent.php]: any reason not to just use the node API here?

**Review by rullzer — CHANGES_REQUESTED** [review]

File: lib/public/Share/Events/VerifyMountPointEvent.php (hunk 1 above)
Comment (nickvergessen) [reply lib/public/Share/Events/VerifyMountPointEvent.php]: In https://github.com/nextcloud/server/pull/19321/files#diff-815375797d10aaf371d625ead4ed15beR93 we don't have the nodes api available and I wanted to keep the impact minimal (remember, we are inside the process of setting up the FS for the user)

Comment (rullzer) [reply lib/public/Share/Events/VerifyMountPointEvent.php]: Right right.
Ok then lets do it.

**Review by rullzer — APPROVED** [review]

**Pull request merged**